document-store/src/integrationTest/java/org/hypertrace/core/documentstore/postgres/PostgresDocStoreTest.java

```diff
@@ -313,750 +201,61 @@ public void testDrop() {
-      Iterator<Document> results = collection.search(query);
-      List<Document> documents = new ArrayList<>();
-      while (results.hasNext()) {
-        documents.add(results.next());
-      }
-      Assertions.assertFalse(documents.isEmpty());
-      String persistedDocument = documents.get(0).toJson();
-      Assertions.assertTrue(persistedDocument.contains("Bob"));
-    }
+    DocStoreTest.testIgnoreCaseLikeQuery(datastore, DocStoreTest.POSTGRES_STORE);
   }
 
   @Test
   public void testInQuery() throws IOException {
-    Collection collection = datastore.getCollection(COLLECTION_NAME);
-    collection.upsert(
-        new SingleValueKey("default", "testKey1"), Utils.createDocument("name", "Bob"));
-    collection.upsert(
-        new SingleValueKey("default", "testKey2"), Utils.createDocument("name", "Alice"));
-    collection.upsert(
-        new SingleValueKey("default", "testKey3"), Utils.createDocument("name", "Halo"));
-
-    List<String> inArray = new ArrayList<>();
-    inArray.add("Bob");
-    inArray.add("Alice");
-
-    Query query = new Query();
-    query.setFilter(new Filter(Filter.Op.IN, "name", inArray));
-    Iterator<Document> results = collection.search(query);
-    List<Document> documents = new ArrayList<>();
-    while (results.hasNext()) {
-      documents.add(results.next());
-    }
-    Assertions.assertEquals(documents.size(), 2);
+    DocStoreTest.testInQuery(datastore);
   }
 
   @Test
   public void testInQueryWithNumberField() throws IOException {
-    Collection collection = datastore.getCollection(COLLECTION_NAME);
-    collection.upsert(
-        new SingleValueKey("default", "testKey1"),
-        Utils.createDocument(
-            ImmutablePair.of("id", "testKey1"),
-            ImmutablePair.of("name", "abc1"),
-            ImmutablePair.of("size", -10.2),
-            ImmutablePair.of("isCostly", false)));
-    collection.upsert(
-        new SingleValueKey("default", "testKey2"),
-        Utils.createDocument(
-            ImmutablePair.of("id", "testKey2"),
-            ImmutablePair.of("name", "abc2"),
-            ImmutablePair.of("size", 10.4),
-            ImmutablePair.of("isCostly", false)));
-    collection.upsert(
-        new SingleValueKey("default", "testKey3"),
-        Utils.createDocument(
-            ImmutablePair.of("id", "testKey3"),
-            ImmutablePair.of("name", "abc3"),
-            ImmutablePair.of("size", 30),
-            ImmutablePair.of("isCostly", false)));
-
-    List<Number> inArray = new ArrayList<>();
-    inArray.add(10.4);
-    inArray.add(30.1);
-
-    Query query = new Query();
-    query.setFilter(new Filter(Filter.Op.IN, "size", inArray));
-    Iterator<Document> results = collection.search(query);
-    List<Document> documents = new ArrayList<>();
-    while (results.hasNext()) {
-      documents.add(results.next());
-    }
-    Assertions.assertEquals(documents.size(), 1);
+    DocStoreTest.testNotInQueryWithNumberField(datastore);
```

Comment:
This test doesn't exit for MongoDocStoreTest, can you enable it and see if works? If, not, don't move it to DocstoreTest. Let's keep those tests in DocStoreTest which works against both.

Reply:
I have added it for Mongo as well 
https://github.com/hypertrace/document-store/pull/41/files#diff-fab258440918057cd0bc0d16483437cc7dbfb88df8e6256b4efb55a763bb9712R538